Get rid of 'protected' (#3162)
modelica/ModelicaSpecification: RationaleMCP/0031/annotations.md

- criterion for selection of which variables to include in the simulation result stored to file

When the Flat Modelica is generated from a full Modelica model, a component coming from a protected section (except inside functions) in the full Modelica model shall have the annotation `Protected = true` in Flat Modelica.
A component coming from a public section must not have the `Protected = true` annotation.

When the Flat Modelica is not generated from a full Modelica model, the `protected` annotation needs to be understood by comparison to the case of full Modelica origin.

For example, consider the following full Modelica model:
```
model M
protected
  parameter Real p = 1.0;
end M;
```

Note that `p` can only be modified when extending the model, and that the `Protected = true` annotation does not enforce this constraint in Flat Modelica.
Hence, the correct conversion to Flat Modelica needs to combine `Protected = true` with treatment similar to a parameter declared `final`:
```
model 'M'
  parameter Real 'p' annotation(Protected = true);
initial equation
  'p' = 1.0; /* From full Modelica protected declaration equation. */
end 'M';
```

> Note that the `Protected = true` annotation doesn't tell at which level of the component hierarchy that the original protected section was, as shown by the following example:

Reviewer: ```suggestion
Note that in full Modelica a protected section can occur at any level of the component hierarchy, as shown by the following example:
```

- `hideResult` — Applied to a parameter or variable, implies the variable should not be included in the simulation output
- `choices` — Applied to a parameter or variable, can be used to enumerate and tag different values for parameter input

These are the new annotations introduced in Flat Modelica, each explained in more detail below:

> - [`protected`](#protected) — Indicate whether component declaration comes from protected section in original full Modelica model

Reviewer: Web Meeting: Hans, Henrik, Gerd, Martin, Oliver

Gerd: Using a Modelica keyword for the annotations doesn't feel right.

Candidate names:

- Protected
- isProtected
